### ISSUE TITLE
Add ability to limit the size of a Select widget dropdown before it scrolls

### ISSUE THREAD

**Issue by dweymouth** [fyne-io/fyne#3800]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same feature, before opening a new one.
- [X] This issue only relates to a single feature. I will open new issues for any other features.

### Is your feature request related to a problem?

Currently a Select widget with a lot of data will make its drop down as large as possible, limited only by the window size, before it adds a scroll bar. This can be awkward for apps running on a large screen with a select placed nowhere near the bottom of the window.

### Describe the solution you'd like to see.

Either change the behavior of Select to show a new hardcoded max number of rows of the dropdown before making it scroll, even if there is room enough in the app window, or expose this through a new API, like a `MaxDropDownRows int` property that if unset (aka zero or negative), leaves the behavior as-is, but if set to a positive value, will show only that number of rows before adding a scrollbar.

**Comment by Jacalz** [comment]
I think the correct fix (or possibly together with this) might be to instead update the popup menu code to use caching logic similar to the List widget. It is currently very slow with large amounts of items.

**Comment by dweymouth** [comment]
As an addendum to this, the dropdown list should never extend above the select widget itself as it currently does if the list is very long.

**Comment by andydotxyz** [comment]
> As an addendum to this, the dropdown list should never extend above the select widget itself as it currently does if the list is very long.

Sadly this does not hold. What is the select is close to the bottom of the screen?

**Comment by Jacalz** [comment]
I believe GTK4 applications on Wayland make the popup appear outside the window in that case but I suspect that it might be hard for us to do (especially across desktops).

**Comment by andydotxyz** [comment]
Sure, in theory that is possible. Unless the window is at the bottom of the screen.

My point was just that it is not sensible to state that the popup must always appear below the select input. There are cases when it may need to be above instead.

**Comment by dweymouth** [comment]
Yes you are correct, I didn't think about the case where it's at the bottom of the window. I still think if there is room for a few rows below, like 5 or so, then the Select should just scroll rather than have the drop-down extend above the widget.

**Comment by andydotxyz** [comment]
I wonder if this is actually platform specific (as, I think, menus would be as well).
On mobile if there is a lot to scroll it would be more common to use the whole screen I think?

**Comment by Bluebugs** [comment]
On mobile you can see the problem in nomad. I think the constrain on mobile will be dependent on the keyboard position on top of where the widget is and the size of the content that we are trying to fit on screen. I don't think we have the ability to read the keyboard position yet, so at the moment, I think they are similar on all platform.
